### PR TITLE
Caplin: tweaks to make staking more stable.

### PR DESCRIPTION
**Pull request by Giulio2002**
Tweaks I did:
1) Decreased attestation expiry down to 30 minutes
2) Removed slot check in committeeSubAggregation
3) More reliable algorithm for the dependent root

Results:
* Better aggregates
* Less strain on the node
* No blocks/attestations missed

### CODE REVIEW

File: cl/beacon/handler/duties_attester.go [hunk 1]
```diff
@@ -22,6 +22,25 @@ type attesterDutyResponse struct {
...
+			dependentRootSlot = 0
+		}
+
+		dependentRoot, err := s.GetBlockRootAtSlot(dependentRootSlot)
+		if err != nil {
+			dependentRootSlot--
+			continue
+		}
+		fmt.Println(dependentRoot)
+		return dependentRoot
```
Comment (domiwei) [inline cl/beacon/handler/duties_attester.go:39]: i feel cannot get the behavior of this function. Is the following code snippet equivalent to your code?

```go
startSlot := ((epoch - 1) * a.beaconChainCfg.SlotsPerEpoch) - 3
tryCount := 0
for slot:=startSlot; slot >=0 && tryCount<2048; slot-- {
  root, err := s.GetBlockRootAtSlot(slot)
  if err == nil {
    return root
  }
  tryCount++
}
return libcommon.Hash{}
```

File: cl/validator/committee_subscription/committee_subscription.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ type CommitteeSubscribeMgmt struct {
 	// subscriptions
 	aggregationPool    aggregation.AggregationPool
 	validatorSubsMutex sync.RWMutex
-	validatorSubs      map[uint64]map[uint64]*validatorSub // slot -> committeeIndex -> validatorSub
```
Comment (domiwei) [inline cl/validator/committee_subscription/committee_subscription.go:43]: curious why don't need slot check anymore.
and i guess we don't need to record `validatorIdxs` in `validatorSub`, right? if so i can remove it and simplify this module in another pr.

File: cl/beacon/handler/duties_attester.go (hunk 1 above)
Comment (Giulio2002) [reply cl/beacon/handler/duties_attester.go:39]: yes, try until works. maybe we miss some blocks

File: cl/beacon/handler/duties_attester.go [hunk 3]
```diff
@@ -22,6 +22,25 @@ type attesterDutyResponse struct {
 	Slot                    uint64            `json:"slot,string"`
 }
 
+func (a *ApiHandler) getDependentRoot(s *state.CachingBeaconState, epoch uint64) libcommon.Hash {
+	dependentRootSlot := ((epoch - 1) * a.beaconChainCfg.SlotsPerEpoch) - 3
+	maxIterations := 2048
+	for i := 0; i < maxIterations; i++ {
+		if dependentRootSlot > epoch*a.beaconChainCfg.SlotsPerEpoch {
+			dependentRootSlot = 0
```
[the remark this answers is not in the document]
Comment (Giulio2002) [reply cl/beacon/handler/duties_attester.go:30]: mmmh okay give me a sec

File: cl/validator/committee_subscription/committee_subscription.go (hunk 2 above)
Comment (Giulio2002) [reply cl/validator/committee_subscription/committee_subscription.go:43]: ok handled it

File: cl/beacon/handler/duties_attester.go [hunk 4]
```diff
@@ -22,6 +22,25 @@ type attesterDutyResponse struct {
 	Slot                    uint64            `json:"slot,string"`
 }
 
+func (a *ApiHandler) getDependentRoot(s *state.CachingBeaconState, epoch uint64) libcommon.Hash {
+	dependentRootSlot := ((epoch - 1) * a.beaconChainCfg.SlotsPerEpoch) - 3
```
[the remark this answers is not in the document]
Comment (Giulio2002) [reply cl/beacon/handler/duties_attester.go:26]: yes, so it might happen with -1 that the block has not arrived yet

**Review by domiwei — APPROVED** [review]

**Pull request merged**